cmd/crio/main.go

```diff
@@ -114,6 +119,7 @@ func main() {
 	// This must be done before reexec.Init()
 	ioutils.SetDefaultOptions(ioutils.AtomicFileWriterOptions{NoSync: true})
 
+	maybeMoveToCgroup()
```

Comment:
I'm confused. this reexec happens in the main crio process. Doesn't that mean we're just moving the whole crio process to a new cgroup?

Reply:
(I should probably learn how reexec.Init() works..)

Reply:
no :) when we do a pull, we re-exec the crio process and the os.Args[0] is set to a special value.  containers/storage does it for running the extraction in a chroot environment (the logic is in `github.com/containers/storage/pkg/chrootarchive/archive_unix.go`).
The current implementation is a hack, to avoid changes in containers/storage and get an idea of what improvements we can expect.